lib/models/ObjectMD.ts

```diff
@@ -892,17 +892,36 @@ export default class ObjectMD {
         return this._data.versionId;
     }
 
+    /**
+     * Get metadata raw versionId value, when object
+     * is in a versioning suspended state it returns 'null'
+     * 
+     * @returns {undefined|string} The object's versionId
+     */
+    getUnencodedVersionId() {
```

Comment:
should this not be in `getVersionId` ?

naively, it seems that having 3 methods to get the versionId is a bit much : so I am thinking we may be missing something here. What are the "use case" for each of the 3 methods?

Reply:
I believe `getVersionId` is a getter for `versionId` with no extra logic, that's why i decided to add a function for our specific case.

`getEncodedVersionId` is used to get the encoded versionId, so the versionId seen by the clients, a null version should never return it's internal versionId, and so here we return `'null'`

`getUnencodedVersionId` has the same logic as the previous but returns the unencoded versionId

Reply:
that's my point : I don't think we need to keep the "accessor with no extra logic" if we don't use it, and it seems we should always either get the encoded or unencoded versionid...

On top of that: there are pieces of code which can be used kind of interchangeably with ObjectMD or other types (KafkaEntry for exemple) : there it expects to have the same method in both types, and I am wondering if that code should benefit from the change "for free"...

Reply:
Looking at where and how the accessor is used, it seems we can indeed add logic to it.

The other types in backbeat all inherit from ObjectMD, so there's nothing to do there. We use ObjectMD function for version logic.